File: py/text_editor.py
import sys

print("Text Editor, version 1.0")
print("Copyright (c) 2020 miniprime1\n")

print("Options")
print("1. Write Mode (w)")
print("2. Read Mode (r)")
print("3. Append Mode (a)")
print("4. Exit")
option = input("Your choice: ")

if option == "1":
    path = input("\nEnter path of file to save: ")
    f = open(path, 'w')
    print("")
    print("[ Enter text to write (To exit, type \"EXIT\"): ]")
    print("=" * 50)
    while True:
        before_data = input()
        if before_data == 'EXIT': break
        data = before_data + "\n"
        f.write(data)
    print("=" * 50)
    f.close

elif option == "2":
    path = input("\nEnter path of file to open: ")
    f = open(path, 'r')
    print("")
    print("=" * 50)
    data = f.read()
    print(data)
    print("=" * 50)
    f.close

elif option == "3":
    path = input("\nEnter path of file to open: ")
    f = open(path, 'a')
    print("")
    print("[ Enter text to append (To exit, type \"EXIT\"): ]")
    print("=" * 50)
    while True:
        before_data = input()
        if before_data == 'EXIT': break
        data = before_data + "\n"
        f.write(data)
    print("=" * 50)
    f.close

elif option == "4":
    sys.exit()

else:
    print("\nError: invalid choice")

File: matlab/Document/TextEditor.m
disp("Text Editor v1.0")
disp("Copyright (c) 2020 miniprime1")
disp(" ")
disp("Options")
disp("1. Write Mode (w)")
disp("1. Read Mode (r)")
disp("3. Append Mode (a)")
option = input("Enter choice: ");
disp(" ")

if (option == 1)
    path = input("Enter path of file to save: ", 's');
    fp = fopen(path, 'w');
    disp(" ")
    disp('[ Enter text to write (To exit, type "EXIT"): ]')
    disp("==================================================")
    while (true)
        data = input("", 's');
        if (data == "EXIT") 
            break 
        end
        fprintf(fp, "%s\n", data);
    end
    disp("==================================================")
    fclose(fp);

elseif (option == 2)
    path = input("Enter path of file to read: ", 's');d
    fp = fopen(path, 'r');
    disp(" ")
    disp("==================================================")
    fprintf("%s\n", fileread(path));
    disp("==================================================")
    fclose(fp);
    
elseif (option == 3)
    path = input("Enter path of file to open/save: ", 's');d
    fp = fopen(path, 'a+');
    disp(" ")
    disp('[ Enter text to write (To exit, type "EXIT"): ]')
    disp("==================================================")
    fprintf("%s\n", fileread(path));
    while (true)
        data = input("", 's');
        if (data == "EXIT") 
            break 
        end
        fprintf(fp, "%s\n", data);
    end
    disp("==================================================")
    fclose(fp);
    
end

clear option
clear path
clear data